packages/central-server/src/index.js

```diff
@@ -28,70 +28,72 @@ import { createApp } from './createApp';
 
 import winston from './log';
 
-/**
- * Set up database
- */
-const database = new TupaiaDatabase();
-const models = new ModelRegistry(database, modelClasses, true);
-
-/**
- * Set up change handlers e.g. for syncing
- */
-if (isFeatureEnabled('MEDITRAK_SYNC_QUEUE')) {
-  createMeditrakSyncQueue(models);
-}
-
-// Pre-cache entity hierarchy details
-const entityHierarchyCacher = new EntityHierarchyCacher(models);
-entityHierarchyCacher.listenForChanges();
-
-// Add listener to refresh analytics table
-const analyticsRefresher = new AnalyticsRefresher(models);
-analyticsRefresher.listenForChanges();
-
-// Add listener to handle survey response changes
-const surveyResponseOutdater = new SurveyResponseOutdater(models);
-surveyResponseOutdater.listenForChanges();
-
-/**
- * Set up actual app with routes etc.
- */
-const app = createApp(database, models);
-
-/**
- * Start the server
- */
-const port = 8090;
-http.createServer(app).listen(port);
-winston.info(`Running on port ${port}`);
-const aggregationDescription = process.env.AGGREGATION_URL_PREFIX || 'production';
-winston.info(`Connected to ${aggregationDescription} aggregation`);
-
-/**
- * Regularly sync data to the aggregation servers
- */
-startSyncWithDhis(models);
-
-/**
- * Regularly sync data to MS1
- */
-startSyncWithMs1(models);
-
-/**
- * Regularly sync data from KoBoToolbox
- */
-startSyncWithKoBo(models);
-
-/**
- * Regularly sync actions that have happened on server with the app social feed.
- */
-startFeedScraper(models);
-
-/**
- * If running via PM2, run migrations then notify that we are ready
- */
-if (process.send) {
-  (async () => {
+(async () => {
+  /**
+   * Set up database
+   */
+  const database = new TupaiaDatabase();
+  const models = new ModelRegistry(database, modelClasses, true);
+
+  /**
+   * Set up change handlers e.g. for syncing
+   */
+  if (isFeatureEnabled('MEDITRAK_SYNC_QUEUE')) {
+    const meditrakSyncQueue = new MeditrakSyncQueue(models);
+    await meditrakSyncQueue.createPermissionsBasedView(); // Ensure permissions based view has been setup
```

Comment:
Always re-create on server startup? Simple: yes, efficient: maybe no... especially since this adds about 30 seconds to the server start time...

Reply:
But good call to go simple to start with, and optimise later if we need to